Classes/Bookmark/BookmarkModel.swift

```diff
@@ -0,0 +1,114 @@
+    }
+    
+    let type: BookmarkType
+    let name: String
+    let owner: String
+    let number: Int
+    let title: String
+    let hasIssueEnabled: Bool
+    
+    init(
+        type: BookmarkType,
+        name: String,
+        owner: String,
+        number: Int = 0,
+        title: String = "",
+        hasIssueEnabled: Bool = false
+    ) {
+        self.type = type
+        self.name = name
+        self.owner = owner
+        self.number = number
+        self.title = title
+        self.hasIssueEnabled = hasIssueEnabled
+    }
+    
+    convenience required init?(coder aDecoder: NSCoder) {
+        guard let owner = aDecoder.decodeObject(forKey: Keys.owner) as? String else { return nil }
+        guard let name = aDecoder.decodeObject(forKey: Keys.name) as? String else { return nil }
+        guard let type = aDecoder.decodeObject(forKey: Keys.type) as? String else { return nil }
+        guard let title = aDecoder.decodeObject(forKey: Keys.title) as? String else { return nil }
+        let hasIssueEnabled = aDecoder.decodeBool(forKey: Keys.hasIssueEnabled)
+        let number = aDecoder.decodeInteger(forKey: Keys.number) as Int
+        
+        self.init(
+            type: BookmarkType(rawValue: type)!,
+            name: name,
+            owner: owner,
+            number: number,
+            title: title,
+            hasIssueEnabled: hasIssueEnabled
+        )
+    }
+    
+    func encode(with aCoder: NSCoder) {
+        aCoder.encode(owner, forKey: Keys.owner)
+        aCoder.encode(number, forKey: Keys.number)
+        aCoder.encode(name, forKey: Keys.name)
+        aCoder.encode(type.rawValue, forKey: Keys.type)
+        aCoder.encode(title, forKey: Keys.title)
+        aCoder.encode(hasIssueEnabled, forKey: Keys.hasIssueEnabled)
+    }
+    
+    override var hashValue: Int {
+        get {
+            return (name + owner + "\(number)").hashValue
+        }
+    }
+    
+    override func isEqual(_ object: Any?) -> Bool {
+        let bookmark = (object as? BookmarkModel)
+        return hashValue == bookmark?.hashValue
+    }
+    
+    // MARK: Filterable
+    
+    func match(query: String) -> Bool {
+        let lowerQuery = query.lowercased()
+        
+        if type.rawValue.contains(lowerQuery) { return true }
+        if String(number).contains(lowerQuery) {return true }
+        if title.lowercased().contains(lowerQuery) { return true }
+        if owner.lowercased().contains(lowerQuery) { return true }
+        if name.lowercased().contains(lowerQuery) { return true }
+        
+        return false
+    }
+    
+    // MARK: Helper
+    
+    func getBookmarkImageIcon() -> UIImage? {
+        var imageName = ""
+        switch type {
+            case .repo: imageName = "repo"
+            case .issue: imageName = "issue-opened"
+            case .pullRequest: imageName = "git-pull-request"
+        }
+        return UIImage.init(named: imageName)?.withRenderingMode(.alwaysTemplate)
```

Comment:
I'm finding this bit of logic is being used in quite a few places around the app now (I know I had to do it for Repo view) - I wonder if there's a nicer interface which enables reuse?